x/marketmap/keeper/keeper.go

```diff
@@ -192,3 +194,32 @@ func (k *Keeper) CreateMarket(ctx sdk.Context, ticker types.Ticker, paths types.
 
 	return k.SetLastUpdated(ctx)
 }
+
+// ValidateState is called after keeper modifications have been made to the market map to verify that
+// the aggregate of all updates has led to a valid state.
+func (k *Keeper) ValidateState(ctx sdk.Context, creates []types.CreateMarket) error {
```

Comment:
small nit, is there a reason this has to take in a list of creates instead of a single one?

Reply:
The `UpdateMarketMap` tx contains a list of creates (so it's sort of batching all of these changes).  We need to verify that for all of these changes, the state is valid